src/docfx/build/xref/XrefResolver.cs

```diff
@@ -230,28 +236,41 @@ private static string RemoveSharingHost(string url, string hostName)
         }
 
         private (Error?, IXrefSpec?, string? href) Resolve(
-            SourceInfo<string> uid, FilePath referencingFile, FilePath inclusionRoot, MonikerList? monikers = null)
+            SourceInfo<string> uid,
+            FilePath referencingFile,
+            FilePath inclusionRoot,
+            bool uIDGlobalUnique,
+            MonikerList? monikers = null)
         {
-            var (xrefSpec, href) = ResolveInternalXrefSpec(uid, referencingFile, inclusionRoot, monikers);
-            if (xrefSpec is null)
-            {
-                (xrefSpec, href) = ResolveExternalXrefSpec(uid);
-            }
+            var (internalXrefSpec, internalHref) = ResolveInternalXrefSpec(uid, referencingFile, inclusionRoot, monikers);
 
-            if (xrefSpec is null)
+            if (uIDGlobalUnique || internalXrefSpec == null)
             {
-                return (Errors.Xref.XrefNotFound(uid), null, null);
+                var (externalXrefSpec, externalHref, repoUrl) = ResolveExternalXrefSpec(uid);
+
+                if (internalXrefSpec == null && externalXrefSpec == null)
+                {
+                    return (Errors.Xref.XrefNotFound(uid), null, null);
+                }
+                else if (internalXrefSpec == null && externalXrefSpec != null)
+                {
+                    return (null, externalXrefSpec, externalHref);
+                }
+                else if (internalXrefSpec != null && externalXrefSpec != null && uIDGlobalUnique)
+                {
+                    return (Errors.Xref.DuplicateUidGlobal(uid, repoUrl), internalXrefSpec, internalHref);
```

Comment:
Sync with @AngryBerryMS on #6599, this PR is doing roughly the same check, so both should check in the same place and consider monikers.

Reply:
OK